plugins/module_utils/jsonpath_extractor.py

```diff
@@ -0,0 +1,142 @@
+            data = value[arr_idx + 1:arr_end]
+            if data != "*" and not data.isnumeric():
+                raise JsonPathException("wrong value specified into array starting at index {0} => '{1}'".format(arr_idx, data))
+            return int(data) if data != "*" else -1, arr_end + 1 if arr_end < len(value) - 1 else None
+        elif arr_idx > start:
+            # single value found (ex: "containers[0]")
+            return value[start:arr_idx], arr_idx
+
+    else:  # dot_idx != -1
+        return value[start:dot_idx], dot_idx + 1
+
+
+def parse(expr):
+    result = []
+    if expr[0] == ".":
+        expr = expr[1:]
+    start = 0
+    while start is not None:
+        elt, next = find_element(expr, start)
+        if elt == '':
+            if not isinstance(result[-1], int):
+                raise JsonPathException("empty element following non array element at index {0} '{1}'".format(start, expr[start:]))
+        else:
+            result.append(elt)
+        start = next
+    return result
+
+
+def search_json_item(jsonpath_expr, json_doc):
+    json_idx = 0
+    json_item = jsonpath_expr[json_idx]
+    if isinstance(json_item, int):
+        if not isinstance(json_doc, list):
+            # trying to parse list items, but current document is not a list
+            return None
+        elements = json_doc
+        if json_item != -1:
+            # looking for specific index from the list
+            if json_item >= len(json_doc):
+                return None
+            else:
+                elements = json_doc[json_item]
+
+        # when we reach the end of the json path
+        if len(jsonpath_expr) == 1:
+            return elements
+        elif json_item != -1 and (isinstance(elements, dict) or isinstance(elements, list)):
+            return search_json_item(jsonpath_expr[1:], elements)
+        elif json_item == -1:
+            result = []
+            for elt in elements:
+                ret = search_json_item(jsonpath_expr[1:], elt)
+                if ret is not None:
+                    result.append(ret)
+            return result if result != [] else None
+    else:
+        # looking for a specific field into the json document
+        if not isinstance(json_doc, dict):
+            return None
+        if json_item not in json_doc:
+            return None
+        if len(jsonpath_expr) == 1:
+            return json_doc.get(json_item)
+        else:
+            return search_json_item(jsonpath_expr[1:], json_doc.get(json_item))
+
+
+def search(expr, data):
+    jsonpath_expr = parse(expr)
+    return search_json_item(jsonpath_expr, data)
+
+
+def validate_with_jsonpath(module, data, expr, value=None):
+    def _raise_or_fail(err, **kwargs):
+        if module and hasattr(module, "fail_json"):
+            module.fail_json(error=to_native(err), **kwargs)
+        raise err
+
+    def _match_value(buf, v):
```

Comment:
This seems broken when trying to wait on a property that is an integer. For example, I can't get the following to work with a Deployment:
```
wait: yes
wait_property:
  property: status.availableReplicas
  value: 1
```

Reply:
updated